modules/sdk-coin-osmo/test/unit/transactionBuilder/StakingActivateBuilder.ts

```diff
@@ -0,0 +1,143 @@
+    const tx = await txBuilder.build();
+    const json = await (await txBuilder.build()).toJson();
+    should.equal(tx.type, TransactionType.StakingActivate);
+    should.deepEqual(json.gasBudget, testTx.gasBudget);
+    should.deepEqual(json.sendMessages, [testTx.sendMessage]);
+    should.deepEqual(json.publicKey, toHex(fromBase64(testTx.pubKey)));
+    should.deepEqual(json.sequence, testTx.sequence);
+    tx.toBroadcastFormat();
+    should.deepEqual(tx.inputs, [
+      {
+        address: testTx.delegator,
+        value: testTx.sendMessage.value.amount.amount,
+        coin: 'tosmo',
+      },
+    ]);
+    should.deepEqual(tx.outputs, [
+      {
+        address: testTx.validator,
+        value: testTx.sendMessage.value.amount.amount,
+        coin: 'tosmo',
+      },
+    ]);
+  });
+
+  it('should sign a Delegate tx', async function () {
+    const txBuilder = factory.getStakingActivateBuilder();
+    txBuilder.sequence(testTx.sequence);
+    txBuilder.gasBudget(testTx.gasBudget);
+    txBuilder.messages([testTx.sendMessage.value]);
+    txBuilder.accountNumber(testTx.accountNumber);
+    txBuilder.chainId(testTx.chainId);
+    txBuilder.sign({ key: toHex(fromBase64(testTx.privateKey)) });
+    const tx = await txBuilder.build();
+    const json = await (await txBuilder.build()).toJson();
+    should.equal(tx.type, TransactionType.StakingActivate);
+    should.deepEqual(json.gasBudget, testTx.gasBudget);
+    should.deepEqual(json.sendMessages, [testTx.sendMessage]);
+    should.deepEqual(json.publicKey, toHex(fromBase64(testTx.pubKey)));
+    should.deepEqual(json.sequence, testTx.sequence);
+    const rawTx = tx.toBroadcastFormat();
+    should.equal(tx.signature[0], toHex(fromBase64(testTx.signature)));
+    should.equal(rawTx, testTx.signedTxBase64);
+    should.deepEqual(tx.inputs, [
+      {
+        address: testTx.delegator,
+        value: testTx.sendMessage.value.amount.amount,
+        coin: 'tosmo',
+      },
+    ]);
+    should.deepEqual(tx.outputs, [
+      {
+        address: testTx.validator,
+        value: testTx.sendMessage.value.amount.amount,
+        coin: 'tosmo',
+      },
+    ]);
+  });
+
+  xit('should submit a send transaction', async () => {
+    const keyPair = new KeyPair({ prv: toHex(fromBase64(testTx.privateKey)) });
+    const axiosConfig = {
+      headers: {
+        'Content-Type': 'application/cbor',
+      },
+      timeout: 10000,
+    };
+
+    const txBuilder = factory.getStakingActivateBuilder();
+    txBuilder.sequence(10);
+    txBuilder.gasBudget(testTx.gasBudget);
+    txBuilder.messages([testTx.sendMessage.value]);
+    txBuilder.accountNumber(testTx.accountNumber);
+    txBuilder.chainId(testTx.chainId);
+    txBuilder.sign({ key: keyPair.getKeys().prv });
+    const tx = await txBuilder.build();
+    const serializedTx = tx.toBroadcastFormat();
+
+    try {
+      const res = await axios.post(
+        'https://lcd.osmotest5.osmosis.zone/cosmos/tx/v1beta1/txs',
```

Comment:
I think, we should mock this API instead of calling again and again. We might hit the rate limit and eventually this is going to fail the testcase.

Reply:
Ideally, we never want to make actual calls in unit tests, so mocking/stubbing should be fine

Reply:
this is the reason for skipping the test. We need to find better way to do this integration test thing